### PR TITLE
Minion 10.12 has native support for expiring jobs

### PR DESCRIPTION
**Pull request by kraih**
This PR replaces the old Gru based TTL feature with the one from Minion upstream. If i'm not mistaken that was the last Gru feature we were depending on that actually required the Gru layer around Minion. Now there is just the `gru_tasks` and `gru_dependencies` tables left. How they are still used is a bit hard to follow, but i wouldn't be surprised if we could just remove them completely. Some dependency management might still be needed. That's something to look into for a followup.

### CODE REVIEW

File: lib/OpenQA/WebAPI/GruJob.pm [hunk 1]
```diff
@@ -21,25 +21,13 @@ use Data::Dumper 'Dumper';
...
-
-    my $err = $self->SUPER::execute;
+    my $gru_id = $self->info->{notes}{gru_id};
+    my $err    = $self->SUPER::execute;
 
     # Non-Gru tasks
     return $err unless $gru_id;
 
-    $info = $self->info;
+    my $info  = $self->info;
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/GruJob.pm:30]: This variable could already be used a few lines above.

Comment (kraih) [reply lib/OpenQA/WebAPI/GruJob.pm:30]: No, it could not. The job metadata changed in the meantime and we need the latest version here.
